pynapple/process/signal_processing.py

```diff
@@ -0,0 +1,267 @@
+"""
+Signal processing tools for Pynapple.
+
+Contains functionality for signal processing pynapple object; fourier transforms and wavelet decomposition.
+"""
+
+from itertools import repeat
+from math import ceil, floor
+
+import numpy as np
+import pandas as pd
+from scipy.signal import welch
+
+import pynapple as nap
+
+
+def compute_spectogram(sig, fs=None):
+    """
+    Performs numpy fft on sig, returns output
+
+    ----------
+    sig : pynapple.Tsd or pynapple.TsdFrame
+        Time series.
+    fs : float, optional
+        Sampling rate, in Hz. If None, will be calculated from the given signal
+    """
+    if not isinstance(sig, (nap.Tsd, nap.TsdFrame)):
+        raise TypeError(
+            "Currently compute_spectogram is only implemented for Tsd or TsdFrame"
+        )
+    if fs is None:
+        fs = sig.index.shape[0] / (sig.index.max() - sig.index.min())
+    fft_result = np.fft.fft(sig.values, axis=0)
+    fft_freq = np.fft.fftfreq(len(sig.values), 1 / fs)
+    return pd.DataFrame(fft_result, fft_freq)
+
+
+def compute_welch_spectogram(sig, fs=None):
+    """
+    Performs scipy Welch's decomposition on sig, returns output
+
+    ----------
+    sig : pynapple.Tsd or pynapple.TsdFrame
+        Time series.
+    fs : float, optional
+        Sampling rate, in Hz. If None, will be calculated from the given signal
+    """
+    if not isinstance(sig, (nap.Tsd, nap.TsdFrame)):
+        raise TypeError(
+            "Currently compute_welch_spectogram is only implemented for Tsd or TsdFrame"
+        )
+    if fs is None:
+        fs = sig.index.shape[0] / (sig.index.max() - sig.index.min())
+    freqs, spectogram = welch(sig.values, fs=fs, axis=0)
+    return pd.DataFrame(spectogram, freqs)
```

Comment:
Ok so wekch returns only the positive freqs so it's better to have the functions match indeed

Reply:
Also I noticed the two functions don't return the same scaled spectrogram